src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs

```diff
@@ -170,7 +171,7 @@ static MsQuicApi()
                 }
                 return;
             }
-            string? gitHash = Marshal.PtrToStringUTF8((IntPtr)libGitHash);
+            string? gitHash = Utf8StringMarshaller.ConvertToManaged((IntPtr)libGitHash);
```

Comment:
```suggestion
            string? gitHash = Utf8StringMarshaller.ConvertToManaged(libGitHash);
```
Also, change the type of `libGitHash` to `byte*`